insights/parsers/engine_config.py

```diff
@@ -0,0 +1,231 @@
+
+    >>> output.head('HotUnplugCpuSupported')
+    '{"x86":"false","ppc":"false"}'
+    >>> output.last('HotUnplugCpuSupported')
+    '{"x86":"true","ppc":"true"}'
+
+``get_version(keyword)`` can be used to retrieve versions(list)::
+
+    >>> output.get_version('HotUnplugCpuSupported')
+    ['3.6', '4.0', '4.1']
+
+For non-exiting keyword::
+
+    >>> 'IDoNotExit' in output
+    False
+    >>> output['IDoNotExit']
+    []
+    >>> output.get('IDoNotExit')
+    []
+    >>> output.get_version('IDoNotExit')
+    []
+    >>> output.head('IDoNotExist)
+    >>> output.last('IDoNotExist)
+"""
+from collections import namedtuple
+from operator import itemgetter
+
+from .. import Parser, parser, get_active_lines
+from insights.specs import Specs
+
+
+@parser(Specs.engine_config_all)
+class EngineConfigAll(Parser):
+    """Parsing output of command ``engine-config --all``
+
+    The parser tries its best to get `value` & `version` for `keyword`
+    specified. It ignores keywords where is fails. At the moment it
+    works well for output which has `keyword`, `value` & `version` in
+    a single line. It skip(rather fails) any `keyword` which has
+    multi-line output.
+
+    Attributes:
+        fields (list): List of `KeyValue` namedtupules for
+                       each line in the configuration file.
+
+        keywords (set): Set of keywords present in the configuration
+                        file, each keyword has been converted to lowercase.
+    """
+    keyvalue = namedtuple('KeyValue',
+                          ['keyword', 'value', 'version', 'kw_lower'])
+    """namedtuple: Represent name value pair as a namedtuple with case."""
+
+    def parse_content(self, content):
+        """Parse each active line for keyword, values & version
+
+        Args:
+
+            content (list): Output of command ``engine-config --all``
+        """
+        self.fields = []
+        for line in get_active_lines(content):
+            try:
+                key, val, ver = itemgetter(0, 1, -1)(line.split(' '))
+                self.fields.append(self.keyvalue(key.strip(':'), val, ver, key.strip(':').lower()))  # noqa
+            except:
+                # TODO: Log an exception
+                pass
+        self.keywords = set([kw.kw_lower for kw in self.fields])
+
+    def __contains__(self, keyword):
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
+        return iter(self.fields)
+
+    def __getitem__(self, keyword):
+        return self.get(keyword)
+
+    def get(self, keyword):
+        """A get value for keyword specified
+
+        Example::
+
+            >>> output.get('MaxStorageVdsTimeoutCheckSec')
+            ['30']
+
+        Args:
+
+            keyword (str): A key. Ex: 'HotUnplugCpuSupported'.
+
+        Returns:
+
+            list: Values associated with a keyword. Returns an empty list if,
+                  1. All the values are empty.
```

Comment:
Better move this two lines forward and keep them the same indent with `list:`. See:  http://www.sphinx-doc.org/en/stable/ext/example_google.html?highlight=Returns

Reply:
Done